src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -244,66 +247,120 @@ public String getUrl() throws IOException {
      *
      * This method provides such an opportunity for subtypes.
      */
-    protected void setActions(List<Action> actions) {
-        this.actions = new CopyOnWriteArrayList<Action>(actions);
+    protected synchronized void setActions(List<Action> actions) {
+            this.actions = new CopyOnWriteArrayList<Action>(actions);
+//            this._wrapper = new ListWrapper(this);
     }
 
-/*
-    We can't use Actionable#actions to store actions because they aren't transient,
-    and we need to store actions elsewhere because this is the only mutable pat of FlowNode.
+    /**
+     * Return the first persistent action on the FlowNode, without consulting TransientActionFactories
+     * Used here because it is much faster to invoke a less-conditional, monomorphic method (50% faster)
+     * @param type Class of action
+     * @param <T>  Action type
+     * @return First persistent action or null if not found
+     */
+    public <T extends Action> T getPersistentAction(@Nonnull Class<T> type) {
```

Comment:
@jglick It could be for now.  I'll do it if it's the only way to get your bee, but still like the idea of being able to invoke it in other performance intensive paths.